Author: Jacob-Vlijm <vlijm@planet.nl>
Last-Update: 2018-10-02
Description: Fixed dropby not popping up.
Applied-Upstream: commit: d648d1f2dcfb9e5a73bfe550b7dd61d4138e8871

---
 budgie-dropby/budgie_dropby.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/budgie-dropby/budgie_dropby.py b/budgie-dropby/budgie_dropby.py
index 9538292..aa6107c 100644
--- a/budgie-dropby/budgie_dropby.py
+++ b/budgie-dropby/budgie_dropby.py
@@ -261,9 +261,12 @@ class BudgieDropByApplet(Budgie.Applet):
             return False

     def scrs_active_check(self):
-        return "screensaver is inactive" in subprocess.check_output([
-            "gnome-screensaver-command", "-q"
-        ]).decode("utf-8")
+        cmd = ["gnome-screensaver-command", "-t"]
+        try:
+            output = subprocess.check_output(cmd).decode("utf-8")
+            return not any(char.isdigit() for char in output)
+        except Exception:
+            return False

     def on_event(self, box, *args):
         if all([
--
2.17.1

